corehq/apps/translations/app_translations/upload_form.py

```diff
@@ -0,0 +1,278 @@
+# coding=utf-8
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import copy
+import re
+from collections import defaultdict
+
+from django.contrib import messages
+from django.utils.translation import ugettext as _
+from lxml import etree
+from lxml.etree import XMLSyntaxError, Element
+
+from corehq.apps.app_manager.exceptions import XFormException
+from corehq.apps.app_manager.models import ShadowForm
+from corehq.apps.app_manager.util import save_xform
+from corehq.apps.app_manager.xform import namespaces, WrappedNode
+from corehq.apps.translations.app_translations.utils import (
+    get_bulk_app_sheet_headers,
+    get_unicode_dicts,
+)
+
+
+
+def update_app_from_form_sheet(app, rows, identifier):
```

Comment:
This is one method :(

Reply:
Split up in a couple of commits: d3885475dc049f74bc1fa31177194ceab059b004 b589442e80342e01722bd18257549456fea28566 251a6320b43a65467bcc4ab855470b5427e9f14b 8e58c1567f6ae6cd1e1de0a6c658a0ef7222f40a

Reply:
Thank you :)